src/libaktualizr-c/libaktualizr-c.cc

```diff
@@ -0,0 +1,80 @@
+#include "libaktualizr-c.h"
+
+Aktualizr *Aktualizr_create(const char *config_path) {
+  Aktualizr *a;
+  try {
+    Config cfg(config_path);
+    a = new Aktualizr(cfg);
+  } catch (const std::exception &e) {
+    std::cerr << "Creating exception: " << e.what() << std::endl;
+    return NULL;
+  }
+  return a;
+}
+
+int Aktualizr_initialize(Aktualizr *a) {
+  try {
+    a->Initialize();
+  } catch (const std::exception &e) {
+    std::cerr << "Initializing exception: " << e.what() << std::endl;
+    return -1;
+  }
+  return 0;
+}
+
+int Aktualizr_uptane_cycle(Aktualizr *a) {
+  try {
+    a->UptaneCycle();
+  } catch (const std::exception &e) {
+    std::cerr << "Uptane cycle exception: " << e.what() << std::endl;
+    return -1;
+  }
+  return 0;
+}
+
+void Aktualizr_destroy(Aktualizr *a) {
+  delete a;
+}
+
+Campaign *Aktualizr_campaign_check(Aktualizr *a) {
+  try {
+    auto r = a->CampaignCheck().get();
+    if (!r.campaigns.empty()) {
+      return new Campaign(r.campaigns[0]);
```

Comment:
Is it fair to assume there will never be multiple campaigns? It would indeed be confusing if there were, but a comment to state the assumption might be worthwhile.

Reply:
It's not supported at the moment, so I think it's worth keeping it simple here. I'll add a comment.